python/ray/serve/tests/test_target_capacity.py

```diff
@@ -373,6 +381,458 @@ def test_autoscaling_scale_to_zero(
+            name="request_signal", namespace=SERVE_NAMESPACE
+        ).remote()
+
+        config = ServeDeploySchema(
+            applications=[
+                ServeApplicationSchema(
+                    name=app_name,
+                    import_path=(
+                        "ray.serve.tests.test_target_capacity:"
+                        "create_autoscaling_controlled_app"
+                    ),
+                    args=dict(
+                        min_replicas=min_replicas,
+                        initial_replicas=initial_replicas,
+                        max_replicas=max_replicas,
+                    ),
+                )
+            ]
+        )
+
+        # Initially deploy at target_capacity 0, and check status.
+        self.apply_config_and_check_status(
+            client,
+            target_capacity=0.0,
+            config=config,
+            app_name=app_name,
+            deployment_name=deployment_name,
+            timeout=20,
+            expected_app_status=ApplicationStatus.RUNNING,
+            expected_deployment_status=DeploymentStatus.HEALTHY,
+            expected_deployment_status_trigger=(
+                DeploymentStatusTrigger.CONFIG_UPDATE_COMPLETED
+            ),
+        )
+        self.check_num_replicas(0, app_name, deployment_name)
+
+        # Increase the target_capacity, and check again.
+        self.apply_config_and_check_status(
+            client,
+            target_capacity=50.0,
+            config=config,
+            app_name=app_name,
+            deployment_name=deployment_name,
+            expected_app_status=ApplicationStatus.DEPLOYING,
+            expected_deployment_status=DeploymentStatus.UPSCALING,
+            expected_deployment_status_trigger=(
+                DeploymentStatusTrigger.CONFIG_UPDATE_STARTED
+            ),
+        )
+        self.check_num_replicas(0, app_name, deployment_name)
+        self.unblock_replica_creation_and_deletion(lifecycle_signal, app_name)
+        self.check_num_replicas(int(0.5 * initial_replicas), app_name, deployment_name)
+
+        # Send requests and check that the application scales up.
+        requests = []
+        handle = serve.get_app_handle(app_name)
+        for _ in range(4 * max_replicas):
+            requests.append(handle.remote())
+        ray.get(lifecycle_signal.send.remote())
+        wait_for_condition(
+            self.check_num_replicas,
+            expected_num_replicas=int(0.5 * max_replicas),
+            app_name=app_name,
+            deployment_name=deployment_name,
+            timeout=20,
+        )
+
+        # Clear requests and check that application scales down.
+        ray.get(request_signal.send.remote())
+        results = [request.result() for request in requests]
+        assert results == ["Hello world!"] * (4 * max_replicas)
+        wait_for_condition(
+            self.check_num_replicas,
+            expected_num_replicas=int(0.5 * initial_replicas),
```

Comment:
Curious (my understanding may be off): why is the number of replicas expected to scale down to 0.5 * 30 instead of 0.5 * 10 after requests have been sent? I'm confused as to why the replicas are limited to `[initial_replicas, max_replicas]` instead of `[min_replicas, max_replicas]` while target capacity is increasing and traffic is being sent. I thought `initial_replicas` was disregarded after the deployment starts autoscaling. Does the meaning of `initial_replicas` change with `target_capacity`?